src_c/polygon.c

```diff
@@ -341,7 +405,64 @@ pg_polygon_copy(pgPolygonObject *self, PyObject *_null)
     return pgPolygon_New(&self->polygon);
 }
 
+static PyObject *
+pg_polygon_move(pgPolygonObject *self, PyObject *const *args, Py_ssize_t nargs)
+{
+    double Dx = 0.0, Dy = 0.0;
+
+    if (nargs == 1) {
+        if (!pg_TwoDoublesFromObj(args[0], &Dx, &Dy)) {
+            goto error;
+        }
+    }
+    else if (nargs == 2) {
+        if (!pg_DoubleFromObj(args[0], &Dx) ||
+            !pg_DoubleFromObj(args[1], &Dy)) {
+            goto error;
+        }
+    }
+    else {
+        goto error;
+    }
+
+    move_polygon(&self->polygon, Dx, Dy);
+    return _pg_polygon_subtype_new2(Py_TYPE(self), self->polygon.vertices,
+                                    self->polygon.verts_num);
```

Comment:
This move function is wrong, you don't move the original polygon, clone it and return it.
What you instead do is (in this case) create a heap allocated double * memory of the correct size, copy the original polygon vertices over, move them and then create the new polygon with that memory. At last you free the original memory.